buildcheck/views/admin_assignments.py

```diff
@@ -1,90 +1,106 @@
-        font_size="sm",
-        on_click=rx.redirect(f"/assign/{id_}"),
+    return rx.dialog.root(
+        rx.dialog.trigger(
+            rx.button(
+                rx.cond(status == "Unassigned", "Assign", "Reassign"),
+                size="2",
+                variant="outline",
+                font_size="sm",
+                on_click=AssignmentState.set_selected_case(id_),
+            )
+        ),
+        assign_dialog()
     )
 
-def render_status(status) -> rx.Component:
-    return rx.cond(
-        status == "Unassigned",
-        rx.badge("Unassigned", color_scheme="gray", variant="soft"),
-        rx.cond(
-            status == "In Review",
-            rx.badge("In Review", color_scheme="blue", variant="soft"),
-            rx.badge("Completed", color_scheme="green", variant="soft")
-        )
+
+def render_status(status: str) -> rx.Component:
+    """Render a status badge based on the assignment status."""
+    if status == "Unassigned":
+        return rx.badge("Unassigned", color_scheme="gray", variant="soft")
+    elif status == "In Review":
+        return rx.badge("In Review", color_scheme="blue", variant="soft")
+    return rx.badge("Completed", color_scheme="green", variant="soft")
+
+
+def assignment_row(item: Assignment) -> rx.Component:
+    """Create a table row for an assignment."""
+    return rx.table.row(
+        rx.table.cell(rx.text(item["id"], font_size="sm")),
+        rx.table.cell(rx.text(item["employee"], font_size="sm")),
+        rx.table.cell(rx.text(item["date"], font_size="sm")),
+        rx.table.cell(
+            rx.link(
+                "View",
+                href=f"/details/{item['id']}",
+                color="blue",
+                font_size="sm",
+                font_weight="medium",
+            )
+        ),
+        rx.table.cell(get_admin_status_tag(item["status"])),
+        rx.table.cell(rx.text(item["reviewer"], font_size="sm")),
+        rx.table.cell(rx.box(action_button(item["status"], item["id"]))),
     )
 
 
 def assignments_table() -> rx.Component:
-    def assignment_row(item):
-        return rx.table.row(
-            rx.table.cell(rx.text(item["id"], font_size="sm")),
-            rx.table.cell(rx.text(item["employee"], font_size="sm")),
-            rx.table.cell(rx.text(item["date"], font_size="sm")),
-            rx.table.cell(
-                rx.link("View", href=f"/details/{item['id']}", color="blue", font_size="sm", font_weight="medium")
-            ),
-            rx.table.cell(get_admin_status_tag(item["status"])),
-            rx.table.cell(rx.text(item["reviewer"], font_size="sm")),
-            rx.table.cell(rx.box(action_button(item["status"], item["id"]))),
-        )
-
     return navbar(), rx.container(
         rx.vstack(
             rx.heading("Blueprint Assignments", size="5", margin_top="1em"),
+            rx.spacer(),
+            rx.text(f"Current User: {UserState.name} ({UserState.role})", font_size="sm", color="gray"),
```

Comment:
```suggestion
            rx.spacer(),
            rx.text(lambda: f"Current User: {UserState.name} ({UserState.role})", font_size="sm", color="gray"),
```

Reply:
this doesn't work since we actually need a computed var. i was trying to avoid that but for demo purposes its fine